sprocs/sync_question_tags.sql

```diff
@@ -0,0 +1,35 @@
+-- Accepts a course ID and a list of objects containins qids and tag IDs,
+-- ensures that all question tags are updated in the DB, and removes
+-- any old unused question tags. 
+
+CREATE OR REPLACE FUNCTION
+    sync_question_tags(
+        IN new_question_tags JSONB
+    ) RETURNS void
+AS $$
+DECLARE
+    question JSONB;
+BEGIN
+    FOR question IN SELECT * FROM JSONB_ARRAY_ELEMENTS(new_question_tags) LOOP
+        WITH new_tags AS (
+            INSERT INTO question_tags (
+                question_id,
+                tag_id,
+                number
+            ) SELECT
+                (question->>0)::bigint,
+                tag_id::bigint,
+                number
+            FROM JSONB_ARRAY_ELEMENTS_TEXT(question->1) WITH ORDINALITY AS t(tag_id, number)
```

Comment:
Why do we need the `_TEXT` variant here? This is generally less efficient, I think?

Reply:
AFAIK, the only way to go from something of type `jsonb` to `bigint` is to go through `text` first, hence this. This is the equivalent of what we do at https://github.com/PrairieLearn/PrairieLearn/blob/master/sprocs/authz_assessment.sql#L34.